[APIS-340] Add comparison of Tx 'value' and estimated fee to balance in EVM sign page

G-Gamja opened the pull request:
- tx의 타입이 `simpleSend`가 아닌 경우에도 tx의 value값이 fee 지불 가능성 체크 로직에 합산될 수 있도록 로직을 수정했습니다.
  - 인앱 Squid EVM 스왑 시 실제 사용되는 코인의 수량(cross chain fee)을 확인할 수 있게 되었습니다.
- 변수명을 가독성이 있도록 수정했습니다.
  -  displayValue -> displayFeeValue
  - totalDisplayAmount -> totalPayableNativeCoinDisplayAmount
  - totalBaseAmount -> totalPayableNativeCoinBaseAmount

G-Gamja committed:
perf((evm-transaction)): check 'value' with balance in evm Transaction page

- 일반 샌드 tx일때만 예상 fee 어마운트에 합산해 예산과 비교했던 로직에서 tx의 value값을 합산하여 계산하도록 수정함

ong-ar commented on src/Popup/pages/Popup/Ethereum/Transaction/entry.tsx:
```diff
@@ -277,10 +277,17 @@ export default function Entry({ queue }: EntryProps) {
   );
 
+  const payableNativeCoinDisplayAmount = useMemo(
+    () => toDisplayDenomAmount(BigInt(toHex(ethereumTx.value || '0x0', { addPrefix: true, isStringNumber: true })).toString(10), decimals),
```
hex 관련 된 변환 function 을 하나 만드는게 좋을 거 같네요. 가독성이 너무 떨어집니다.

ong-ar replied:
`ethereumTx.value` 를 하나의 변수로 만드는 것도 좋을 것 같네요.

G-Gamja replied:
toHex와 10진수 변환 로직을 합한 유틸 함수를 추가했습니다. 
또 ethereumTx.value값을 별도로 `nativeCoinTransferBaseAmount`로 선언하도록 코드를 수정했습니다.

* 변수의 이름 가독성 개선을 위해 payable native coin을 native coin transfer의 맥락으로 변경했습니다. 확인 부탁드립니다.

G-Gamja committed:
feat((util)): Add hexOrDecimalToDecimal & rename vars

- 파람을 16,10진수의 값을 받아 10진수의 값을 반환하는 유틸 함수 추가
- 변수 명 변경
  - payable native coin -> native coin transfer의 맥락으로 변경

ong-ar reviewed (approved):
고생하셨습니다.
